File: 30_of_codes/Day 14 - Scope/14.py
class Difference:
    def __init__(self, a):
        self.__elements = a

    # Add your code here
    def computeDifference(self):
        maximumValue = 0
        for i in range(len(self.__elements)):
            for j in range(len(self.__elements)):
                absoluteValue = abs(self.__elements[i] - self.__elements[j])
                if absoluteValue > maximumValue:
                    maximumValue = absoluteValue
        self.maximumDifference = maximumValue


# End of Difference class

_ = input("Number of arrays")
a = [int(e) for e in input().split(' ')]

d = Difference(a)
d.computeDifference()

print(d.maximumDifference)

File: 30_of_codes/Day 14 - Scope/14.swift
import Foundation

class Difference {
    private var elements = [Int]()
    var maximumDifference: Int
    init(a: [Int]) {
        self.elements = a
        self.maximumDifference = 0
    }

    func computeDifference() {
        // Satisfies constraint specified in project
        guard elements.count >= 1, elements.count <= 100 else { return }
        // Sort the array in ascending order
        elements.sort{$0 < $1}
        // Compute the difference between greatest and least in the array.
        maximumDifference = elements.last! - elements.first!
    }

} // End of Difference class

let n = Int(readLine()!)!
let a = readLine()!.components(separatedBy: " ").map{ Int($0)! }

let d = Difference(a: a)

d.computeDifference()

print(d.maximumDifference)
